src/slave/containerizer/mesos/isolators/xfs/disk.cpp

```diff
@@ -464,6 +467,38 @@ Future<Nothing> XfsDiskIsolatorProcess::recover(
     }
   }
 
+  // Assign project IDs to sandboxes that were previously not managed by this
+  // isolator. Quotas will be set later when the containerizer will send an
+  // update call upon executor re-registration.
+  foreach (const string& sandbox, unmanaged) {
+    ContainerID containerId;
+    containerId.set_value(Path(sandbox).basename());
+    CHECK(!infos.contains(containerId));
+
+    // Ignore the container if it is not alive or if it is a known orphan. In
+    // the latter case the containerizer will send a cleanup call for it anyway.
+    if (orphans.contains(containerId) || !alive.contains(containerId)) {
+      continue;
+    }
+
+    Option<prid_t> projectId = nextProjectId();
+    if (projectId.isNone()) {
+      return Failure("Failed to assign project ID, range exhausted");
```

Comment:
"Failed to assign project to sandbox <sandbox>: Failed to obtain next project ID: range exhausted" (wrap lines at 80 though, so split onto multiple lines).